src/main/java/com/microsoft/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -0,0 +1,104 @@
+package com.microsoft.cse.helium.app.services.keyvault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Value;
+import java.io.IOException;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.credentials.MSICredentials;
+import com.microsoft.azure.credentials.AzureCliCredentials;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.models.CertificateBundle;
+import com.microsoft.azure.keyvault.models.KeyBundle;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+
+@Service
+public class KeyVaultService implements IKeyVaultService
+{
+    private String _keyVaultName = "";
```

Comment:
Any particular reason you are initializing them to empty when they are going to get overwritten later? Probably better to hit a "NullPointerException" than use the empty string. Makes it a bit clearer too.

Reply:
It was initially unassigned.  However, I got a linter warning about unassigned which I'm no longer getting.  The class can't be instantiated without matched params which are immediately assigned and validated.  So, it doesn't much matter whether it is null, empty, or some random string.